### ISSUE TITLE
Self-types in non-native JS traits cause confusing error message 

### ISSUE THREAD

**Issue by gzm0** [scala-js/scala-js#3998]
```scala
trait Foo extends js.Object { self => }
```

Causes the following error (without position!).

```
 Members of non-native JS traits must either be abstract, or their right-hand-side must be `js.undefined`.
```

**Comment by gzm0** [comment]
This also happens in both 1.0.x and 0.6.x

**Comment by gzm0** [comment]
I can't figure out how to recognize a self-type symbol :-/

**Comment by sjrd** [comment]
It's called `classSymbol.info.thisType` in scalac. But it's just a type, not a symbol.

**Comment by gzm0** [comment]
Fixed in a176fefe6a5570928480c1c8159972b6cba804d5

**Comment by aappddeevv** [comment]
Thanks!
